.github/workflows/ci.yml

```diff
@@ -0,0 +1,307 @@
+          JRELEASER_MAVENCENTRAL_USERNAME : ${{ secrets.JRELEASER_MAVENCENTRAL_USERNAME }}
+          JRELEASER_MAVENCENTRAL_PASSWORD : ${{ secrets.JRELEASER_MAVENCENTRAL_PASSWORD }}
+          BUILD_RELEASE: true          
+
+  dockerTests:
+    needs: 
+      - assemble
+      - unitTests
+    # 4 cpu, 16G ram
+    runs-on: ubuntu-24.04
+    environment: docker
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: distribution
+          path: build/distributions/
+
+      - name: Docker Metadata
+        id: docker-metadata
+        uses: ConsenSys/github-actions/java-docker-metadata@main
+        with:
+          publish-version: ${{ needs.assemble.outputs.publish-version }}
+          docker-repo: "consensys/linea-shomei"
+
+      - name: Build and export to docker
+        uses: docker/build-push-action@v6
+        env:
+          DOCKER_BUILD_SUMMARY: false
+        with:
+          file: docker/openjdk-21/Dockerfile
+          context: .
+          build-args: |
+            TAR_FILE=./build/distributions/shomei-${{ needs.assemble.outputs.publish-version }}.tar.gz
+            BUILD_DATE=${{ steps.docker-metadata.outputs.build-date }}
+            VCS_REF=${{ steps.docker-metadata.outputs.vcs-ref }}
+            VERSION=${{ needs.build.outputs.publish-version }}
+          no-cache: true
+          load: true
+          tags: consensys/linea-shomei:test
+
+      - name: Get absolute path of reports directory
+        id: get-reports-dir
+        run: echo "path=$(realpath ./build/reports)" >> $GITHUB_OUTPUT
+
+      - name: Run Docker tests
+        run: ./docker/test.sh 'consensys/linea-shomei:test' 
+
+      - name: Test Summary
+        if: always()
+        run: |
+          SUMMARY_CONTENT="<h2>Docker Test Summary</h2>\n"
+          SUMMARY_CONTENT+="<details><summary><strong>Details</strong></summary>\n"
+          SUMMARY_CONTENT+="<pre><code>\n"
+          SUMMARY_CONTENT+=$(cat ./docker/reports/*.xml)
+          SUMMARY_CONTENT+="\n</code></pre></details>\n"
+          echo -e "$SUMMARY_CONTENT" >> $GITHUB_STEP_SUMMARY  
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@0.29.0
+        with:
+          image-ref: 'consensys/linea-shomei:test'
+          format: 'sarif'
+          output: 'trivy-results.sarif'
+          vuln-type: 'os,library'
+          severity: 'CRITICAL,HIGH'
+
+      - name: Upload Trivy scan results to GitHub Security tab
+        uses: github/codeql-action/upload-sarif@v3
+        if: always()
+        with:
+          sarif_file: 'trivy-results.sarif'
+
+  dockerPublish:
+    needs: 
+      - assemble
```

Comment:
also possibly tag, if we want to use tag based release

Reply:
I'd prefer tag but was under the impression that linea uses snapshot versions as well?